spec/sendcloud/operations/cancel_parcel_request_spec.rb

```diff
@@ -0,0 +1,81 @@
+# frozen_string_literal: true
+require "spec_helper"
+
+RSpec.describe Sendcloud::Operations::CancelParcelRequest do
+  let(:default_base_url) { "https://panel.sendcloud.sc/api/v2/" }
+
+  before { configure_client(base_url: default_base_url) }
+
+  describe '#execute' do
+    subject { described_class.new(parcel_id: parcel_id) }
+
+    context "with invalid parcel id" do
+      let(:parcel_id) { "0" }
+
+      it "raises a not found exception" do
+        VCR.use_cassette("cancel_parcel_request/invalid_id") do
+          aggregate_failures do
+            expect {
+              subject.execute
+            }.to raise_error(
+              Sendcloud::Errors::ResponseError, "404 No Parcel matches the given query."
+            )
+            expect(subject.response.status).to eq(404)
+          end
+        end
+      end
+    end
+
+    context "when parcel is valid" do
+      let(:parcel_id) { "124320972" }
+
+      it "cancels the parcel" do
+        VCR.use_cassette("cancel_parcel_request/valid_parcel") do
+          result = subject.execute
+
+          aggregate_failures do
+            expect(subject.response.status).to eq(200)
+            expect(result).to match(hash_including(
+              message: "Parcel has been cancelled",
+              status: "cancelled"
+            ))
+          end
+        end
+      end
+    end
+
+    context "when parcel is deleted" do
+      let(:parcel_id) { "124309702" }
+
+      it "raise deleted error" do
+        VCR.use_cassette("cancel_parcel_request/deleted_parcel") do
+          aggregate_failures do
+            expect {
+              subject.execute
+            }.to raise_error(
+              Sendcloud::Errors::DeletedError, "410 deleted Parcel has been deleted"
+            )
+            expect(subject.response.status).to eq(410)
+          end
+        end
```

Comment:
As above, not sure if this is what we'd want or not, in terms of throwing an error rather than passing the message that it was successfully deleted. Maybe missing something.